openmdao/solvers/solver.py

```diff
@@ -591,13 +592,30 @@ def _solve(self):
         self._mpi_print(self._iter_count, norm, norm / norm0)
 
         stalled = False
+        stall_count = 0
         if stall_limit > 0:
-            stall_count = 0
             stall_norm = norm0
 
         while self._iter_count < maxiter and norm > atol and norm / norm0 > rtol and not stalled:
             with Recording(type(self).__name__, self._iter_count, self) as rec:
-                self._single_iteration()
+
+                if stall_count == 3 and not self.linesearch.options['print_bound_enforce']:
+
+                    self.linesearch.options['print_bound_enforce'] = True
+
+                    msg = ("Your model has stalled three times and may be violating the "
+                           "bounds. In the future, turn on print_bound_enforce in your "
+                           "solver options to see. For example:\n"
+                           "'prob.model.nonlinear_solver.linesearch.options"
```

Comment:
I think the intent of Issue #1811 was that the actual path to _this_ solver is included in the warning:

> The warning should include the full path of the solver, like this: <something>.<something>.<something>.<something>.nonlinear_solver.linesearch.options['print_bound_enforce']=True.